java/code/src/com/redhat/rhn/frontend/xmlrpc/recurringaction/RecurringActionHandler.java

```diff
@@ -0,0 +1,243 @@
+     *
+     * @param loggedInUser The current user
+     * @param actionProps Map containing action properties
+     * @return action id or exception thrown otherwise
+     *
+     * @xmlrpc.doc Create a new recurring action.
+     * @xmlrpc.param #session_key()
+     * @xmlrpc.param
+     *  #struct("actionProps")
+     *      #prop_desc("string", "entity_type", "The type of the target entity")
+     *      #prop_desc("string", "entity_id", "The id of the target entity")
+     *      #prop_desc("string", "name", "The name of the action")
+     *      #prop_desc("string", "cron_expr", "The execution frequency of the action")
+     *      #prop_desc("boolean", "test", "Whether the action should be executed in test mode")
+     *  #struct_end()
+     * @xmlrpc.returntype int action_id - The action id of the recurring action
+     */
+    public Long create(User loggedInUser, Map<String, Object> actionProps) {
+        RecurringAction action = createAction(actionProps, loggedInUser);
+        return save(loggedInUser, action);
+    }
+
+    /* Helper method */
+    private RecurringAction createAction(Map<String, Object> actionProps, User user) {
+        RecurringAction action;
+        if (actionProps.containsKey("id") || !actionProps.containsKey("entity_type") ||
+                !actionProps.containsKey("entity_id") || !actionProps.containsKey("cron_expr") ||
+                !actionProps.containsKey("name")) {
+            throw new InvalidArgsException("Incomplete action props");
+        }
+        action = RecurringActionManager.createRecurringAction(
+                getEntityType((String) actionProps.get("entity_type")),
+                Long.parseLong((String) actionProps.get("entity_id")),
+                user
+        );
+        action.setName((String) actionProps.get("name"));
+        action.setCronExpr((String) actionProps.get("cron_expr"));
+        if (actionProps.containsKey("test")) {
+            action.setTestMode(Boolean.parseBoolean(actionProps.get("test").toString()));
+        }
+        return action;
+    }
+
+    /**
+     * Update a recurring action.
+     *
+     * @param loggedInUser The current user
+     * @param actionProps Map containing properties to update
+     * @return action id or exception thrown otherwise
+     *
+     * @xmlrpc.doc Update a recurring action.
+     * @xmlrpc.param #session_key()
+     * @xmlrpc.param
+     *  #struct("actionProps")
+     *      #prop_desc("string", "id", "The id of the action to update")
+     *      #prop_desc("string", "name", "The name of the action (optional)")
+     *      #prop_desc("string", "cron_expr", "The execution frequency of the action (optional)")
+     *      #prop_desc("boolean", "test", "Whether the action should be executed in test mode (optional)")
+     *      #prop_desc("boolean", "active", "Whether the action should be active (optional)")
+     *  #struct_end()
+     * @xmlrpc.returntype int action_Id - The action id of the recurring action
```

Comment:
The `action_Id` looks a bit weird. But anyway: neither type, nor name is needed here.
```suggestion
     * @xmlrpc.returntype The id of the recurring action
```